app/core/explorer.js

```diff
@@ -1,77 +1,41 @@
 // @flow
 import { openExternal } from './electron'
-import { EXPLORERS } from '../core/constants'
-import { isMainNetwork } from '../core/networks'
-
-export const getExplorerBaseURL = (networkId: string, explorer: ExplorerType) => {
-  let baseURL
-  if (explorer === EXPLORERS.NEO_TRACKER) {
-    if (isMainNetwork(networkId)) {
-      baseURL = 'https://neotracker.io'
-    } else {
-      baseURL = 'https://testnet.neotracker.io'
-    }
-  } else if (explorer === EXPLORERS.NEO_SCAN) {
-    if (isMainNetwork(networkId)) {
-      baseURL = 'https://neoscan.io'
-    } else {
-      baseURL = 'https://neoscan-testnet.io'
-    }
-  } else {
-    if (isMainNetwork(networkId)) {
-      baseURL = 'http://antcha.in'
-    } else {
-      baseURL = 'http://testnet.antcha.in'
-    }
+import explorerConfig from './explorerConfig'
+import { isMainNetwork } from './networks'
+
+export const buildExplorerLink = (
+  networkId: string,
+  explorer: ExplorerType,
+  structure: string,
+  suffix: string
+) => {
+  if (Object.prototype.hasOwnProperty.call(explorerConfig, explorer)) {
```

Comment:
Why is this needed? won't `if (!!explorerConfig[explorer]) { ... }` suffice?

Reply:
My personal feeling on !! is that's it's an anti-pattern. They do a similar thing in the end but I feel that the obj.hasOwnProperty code is more clear. The more verbose version is required because this is React.

Reply:
How about `explorerConfig.hasOwnProperty(explorer)` or even `(explorer in explorerConfig)`?

Reply:
`explorerConfig.hasOwnProperty(explorer)` is not React compatible, which is why I used the longer syntax.

`in` will also return true if key gets found somewhere in the prototype chain, whereas Object.hasOwnProperty (like the name already tells us), will only return true if key is available on that object directly (its "owns" the property). [Citation](https://stackoverflow.com/questions/13632999/if-key-in-object-or-ifobject-hasownpropertykey)

Reply:
What do you mean its not React compatible?

If you are doing a serious runtime validation, then your check is not sufficient. What would it help to know if the key just exists there? It can still fail if it doesn't adhere to a schema.
I recommend to use a schema validation, such as https://github.com/hapijs/joi. You can also write a flow type for `explorerConfig` to describe the shape.

Reply:
I had issues using obj.hasOwnProperty directly in React 15. They must have changed something about how objects are built for 16, but that means obj.hasOwnProperty works now.